### PR TITLE
stdlib: Declare _UnmanagedOpaqueString.UnicodeScalarIterator fixed layout

### PR DESCRIPTION
**Pull request by slavapestov**

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@milseman I think we need this to recover lost performance when resilience is enabled. Is this OK?

File: stdlib/public/core/UnmanagedOpaqueString.swift [hunk 1]
```diff
@@ -340,6 +340,7 @@ extension _UnmanagedOpaqueString : _StringVariant {
   }
 
   @_versioned // FIXME(sil-serialize-all)
+  @_fixed_layout
```
Comment (jrose-apple) [inline stdlib/public/core/UnmanagedOpaqueString.swift]: Please leave the FIXME on any fixed-layouts you add. Only the stdlib team should be able to add them without FIXMEs.

**Comment by milseman** [comment]
@slavapestov maybe, do you have any metrics that demonstrate this?

@swift-ci please smoke benchmark

**Comment by slavapestov** [comment]
@swift-ci Please smoke test Linux

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

File: stdlib/public/core/UnmanagedOpaqueString.swift (hunk 1 above)
Comment (slavapestov) [reply stdlib/public/core/UnmanagedOpaqueString.swift]: Done

**Review by milseman — APPROVED** [review]
LGTM

**Pull request merged**